### PR TITLE
ability to write matml as string

### PR DESCRIPTION
**Pull request by koubaa**
the writer only supports writing to file. I would like to support writing to string in order to avoid having to write a temp file for the onscale integration

### CODE REVIEW

File: src/ansys/materials/manager/util/matml/matml_from_material.py [hunk 1]
```diff
@@ -193,11 +179,47 @@ def export(self, path: _PATH_TYPE, **kwargs):
...
+        """
+        Write a Matml (engineering data xml format) representation of materials to buffer.
+
+        Parameters
+        ----------
+        buffer:
+            buffer to write to
+        Keyword arguments
+        -----------------
+        indent - add an indent to format the XML output (python 3.9+). Defaults to false.
```
Comment (Revathyvenugopal162) [inline src/ansys/materials/manager/util/matml/matml_from_material.py:200]:
```suggestion
        **kwargs : bool, optional
            Optional keyword arguments.
            indent : bool, optional
                Whether to add an indent to format the XML output(python 3.9+). Defaults to ``false``.
```

File: src/ansys/materials/manager/util/matml/matml_from_material.py [hunk 2]
```diff
@@ -193,11 +179,47 @@ def export(self, path: _PATH_TYPE, **kwargs):
...
+        """
+        tree = self._to_etree()
+
+        if kwargs.get("indent", False):
```
Comment (da1910) [inline src/ansys/materials/manager/util/matml/matml_from_material.py:204]: Can you parameterize the read/write test with indent as a bool to cover these two missing hunks?

**Review by da1910 — CHANGES_REQUESTED** [review]
Some of the missing lines are going to be a pain to hit, but the indent hunks would be good to get since we explicitly warn the user if they don't have support.

**Comment by da1910** [comment]
> @da1910 We could make the CI/CD testing matrix include python <3.9 to fully cover indent in codecov, but I'd say it's not worth the trouble.

Agreed, at some point I can add `minidom` as a way to indent code on python 3.8 but for the moment it's fine as-is. Just wanted to cover the indent method in some form.

**Review by da1910 — APPROVED** [review]
This looks OK to me

**Review by Revathyvenugopal162 — APPROVED** [review]

**Pull request merged**